File: tools/df-save-re/ghidra_scripts/decompile_targets.py
# Ghidra postScript: decompile functions at supplied addresses (hex strings in args).
# @category DF

from ghidra.app.decompiler import DecompInterface

if len(getScriptArgs()) == 0:
    printf("usage: decompile_targets.py <addr>...\n")
else:
    ifc = DecompInterface()
    ifc.openProgram(currentProgram)
    for arg in getScriptArgs():
        addr = toAddr(arg)
        fn = getFunctionContaining(addr)
        if fn is None:
            printf("no function at %s\n" % arg)
            continue
        res = ifc.decompileFunction(fn, 30, monitor)
        printf("=== %s ===\n%s\n" % (fn.getName(), res.getDecompiledFunction().getC()))

File: tools/df-save-re/ghidra_scripts/DecompileTargets.java
// Ghidra postScript: decompile functions at addresses passed as args.
//@category DF

import ghidra.app.decompiler.DecompInterface;
import ghidra.app.decompiler.DecompileResults;
import ghidra.app.script.GhidraScript;
import ghidra.program.model.address.Address;
import ghidra.program.model.listing.Function;

public class DecompileTargets extends GhidraScript {
    @Override
    public void run() throws Exception {
        String[] args = getScriptArgs();
        if (args.length == 0) {
            println("usage: DecompileTargets.java <addr>...");
            return;
        }

        DecompInterface ifc = new DecompInterface();
        ifc.openProgram(currentProgram);

        for (String arg : args) {
            Address addr = toAddr(arg);
            Function fn = getFunctionContaining(addr);
            if (fn == null) {
                println("no function at " + arg);
                continue;
            }
            DecompileResults res = ifc.decompileFunction(fn, 60, monitor);
            println("=== " + fn.getName() + " @ " + fn.getEntryPoint() + " ===");
            if (res.decompileCompleted() && res.getDecompiledFunction() != null) {
                println(res.getDecompiledFunction().getC());
            } else {
                println("decompile failed: " + res.getErrorMessage());
            }
        }
    }
}
